### PR TITLE
UCT/IB: Set default traffic class for RoCEv2, and some code cleanups

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -930,11 +937,18 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
...
+                                     self->config.gid_index, &self->gid,
+                                     &is_roce_v2);
     if (status != UCS_OK) {
         goto err;
     }
 
+    if (is_roce_v2 && (config->traffic_class == UCS_CONFIG_ULUNITS_AUTO)) {
+        self->config.traffic_class = UCT_IB_DEFAULT_ROCEV2_DSCP;
+    } else {
+        self->config.traffic_class = config->traffic_class;
```
Comment (brminich) [inline src/uct/ib/base/ib_iface.c]: what if `is_roce_v2` is `false`, but `traffic_class` is `UCS_CONFIG_ULUNITS_AUTO`?

File: test/gtest/uct/ib/test_ib.cc [hunk 2]
```diff
@@ -212,7 +212,7 @@ class test_uct_ib : public uct_test {
 
         uct_ib_address_unpack(ib_addr, &lid_out, &gid_out);
 
-        if (IBV_PORT_IS_LINK_LAYER_ETHERNET(uct_ib_iface_port_attr(iface))) {
+        if (uct_ib_iface_is_roce(iface)) {
```
Comment (brminich) [inline test/gtest/uct/ib/test_ib.cc]: minor: there is another `IBV_PORT_IS_LINK_LAYER_ETHERNET` on line 167

File: src/uct/ib/base/ib_iface.c [hunk 3]
```diff
@@ -1011,6 +1024,9 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
         self->is_global_addr = 0;
     }
 
+    self->config.traffic_class      = config->traffic_class;
```
Comment (brminich) [inline src/uct/ib/base/ib_iface.c]: you already set it on line [946](https://github.com/openucx/ucx/pull/3061/files#diff-694c9dba586b3d5c9d63a14795983496R946)

File: test/gtest/uct/ib/test_ib.cc (hunk 2 above)
Comment (yosefe) [reply test/gtest/uct/ib/test_ib.cc]: i can't change the one in line 167 because it gets port_attr and not ib_iface

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8330/ for details (Mellanox internal link).

**Comment by brminich** [comment]
bot:mlx:retest

**Pull request merged**